### PR TITLE
Adding fill progress bar.

### PR DESCRIPTION
**Pull request by biocom**
Adds a fill progress bar for the Filled section. Created a common component 'progressBar' to be re-used:

Screen Shot 2021-03-02 at 13 39 37
Screen Shot 2021-03-02 at 13 39 30

https://pr235--gpui.review.gnosisdev.com/storybook/?path=/story/common-progressbar--default
https://pr235--gpui.review.gnosisdev.com/storybook/?path=/story/orders-filledprogress--default

### CODE REVIEW

**Comment by biocom** [comment]
Adressing #68

**Review by alfetopito — APPROVED** [review]
Great work, looks amazing!

File: src/components/common/ProgressBar/index.tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    margin: 0 0 0 0.7rem;
+  }
+`
+
+export function ProgressBar(props: Props): JSX.Element {
+  const { percentage = '0', showLabel = true } = props
+
+  return (
+    <Wrapper {...props}>
+      <div>
```
Comment (biocom) [inline src/components/common/ProgressBar/index.tsx:44]: I kinda like having a compact styled component with very specific nested selectors (> b { ....)

But in terms of readability, would it be preferred to turn the nested elements into it's own styled components? E.g.

<div> --> OuterBar
<span> --> InnerBar

?

File: src/components/common/ProgressBar/index.tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  }
+
+  > b {
+    color: ${({ activeColor, theme }): string => (activeColor ? activeColor : theme.green)};
+    margin: 0 0 0 0.7rem;
+  }
+`
+
+export function ProgressBar(props: Props): JSX.Element {
+  const { percentage = '0', showLabel = true } = props
```
Comment (biocom) [inline src/components/common/ProgressBar/index.tsx:40]: Providing defaults, given we have optional props. Preferred way to do it this way?

File: src/components/common/ProgressBar/index.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  > div {
+    width: 16rem;
+    height: 0.6rem;
+    position: relative;
+    border-radius: 16rem;
+    background: ${({ theme }): string => theme.borderPrimary};
+  }
+
+  > div > span {
+    width: ${({ percentage }): string => percentage || '0'}%;
```
Comment (biocom) [inline src/components/common/ProgressBar/index.tsx:23]: How to keep the fallback out of this here? Can only be done in the parent component were we get the props from?

File: src/components/common/ProgressBar/index.tsx (hunk 1 above)
Comment (anxolin) [reply src/components/common/ProgressBar/index.tsx:44]: I'm happy with both approaches. By the size of this component, your approach is good. 
Probably u are right that using  it's own styled components is more readable, but it's kind of fine in this specific case.

File: src/components/orders/FilledProgress/FilledProgress.stories.tsx [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+} as Meta
+
+const Template: Story<Props> = (args) => <FilledProgress {...args} />
+
+const defaultArgs: Props = {
+  percentage: '40',
+}
+
+export const Default = Template.bind({})
+Default.args = { ...defaultArgs }
```
Comment (anxolin) [inline src/components/orders/FilledProgress/FilledProgress.stories.tsx:23]: nit picking, I would suggest to add another two stories: 0% and 100%

**Review by anxolin — APPROVED** [review]
Looks great!

File: src/components/common/ProgressBar/index.tsx [hunk 5]
```diff
@@ -0,0 +1,50 @@
+import React from 'react'
+import styled from 'styled-components'
+
+export type Props = {
+  readonly percentage?: string
+  readonly activeColor?: string
+  readonly showLabel?: boolean
+}
+
+const Wrapper = styled.div<Props>`
```
Comment (anxolin) [inline src/components/common/ProgressBar/index.tsx:10]: You only do this if it's true that you use all the props in the styled-component, but for example `showLabel` is not used

It's more declarative if you use the ones you need. And then, for example, you would be able to define `percentage` as mandatory, so you can do the `percentage || 0` in the reder function, not in the CSS

File: src/components/common/ProgressBar/index.tsx (hunk 3 above)
Comment (anxolin) [reply src/components/common/ProgressBar/index.tsx:23]: You can define the props of the styled compoenent different to the component itself. See my other comment

File: src/components/common/ProgressBar/index.tsx (hunk 2 above)
Comment (anxolin) [reply src/components/common/ProgressBar/index.tsx:40]: yes, perfect

File: src/components/common/ProgressBar/index.tsx (hunk 1 above)
Comment (alfetopito) [reply src/components/common/ProgressBar/index.tsx:44]: I personally prefer smaller styled components, but am ok with this approach too.

File: src/components/common/ProgressBar/index.tsx (hunk 5 above)
Comment (biocom) [reply src/components/common/ProgressBar/index.tsx:10]: Makes sense, thank you.

**Pull request merged**